### PR TITLE
feat: allow gov module account to transfer any CL position

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #7329 

## What is the purpose of the change

Adds logic to allow the governance module account to transfer any position.

## Testing and Verifying

Gotest added.

## Summary by CodeRabbit

- **New Features**
	- Enhanced the concentrated liquidity functionality to allow the governance module account to transfer liquidity positions, improving system flexibility and control.
- **Tests**
	- Updated tests to support the new feature of governance account transfers.

### CODE REVIEW

File: x/concentrated-liquidity/position_test.go [hunk 1]
```diff
@@ -2279,6 +2280,7 @@ func (s *KeeperTestSuite) TestTransferPositions() {
 		positionsToTransfer  []uint64
 		setupUnownedPosition bool
 		isLastPositionInPool bool
+		isGovAddress         bool
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply x/concentrated-liquidity/position_test.go:2283]: This just isn't true...

File: x/concentrated-liquidity/position_test.go [hunk 2]
```diff
@@ -2314,6 +2316,13 @@ func (s *KeeperTestSuite) TestTransferPositions() {
 			outOfRangePositions: []uint64{DefaultPositionId + 1, DefaultPositionId + 2},
 			positionsToTransfer: []uint64{DefaultPositionId, DefaultPositionId + 2},
 		},
+		"three position IDs, not an owner of one of them but caller is gov address": {
```
Comment (p0mvn) [inline x/concentrated-liquidity/position_test.go:2319]: Since this is a gov module acc as the sender, it owns none of them. Is it not?

**Review by p0mvn — APPROVED** [review]

**Review by PaddyMc — APPROVED** [review]

**Comment by ValarDragon** [comment]
I don't really like this, though it does solve the immediate problem.

I think its ok to ship this, but we need to add an issue to correct the ownership structure of this. (Community pool gov module gets direct control of all accounts it controls, e.g. via smart accounts like logic)

**Comment by czarcas7ic** [comment]
> I don't really like this, though it does solve the immediate problem.
> 
> I think its ok to ship this, but we need to add an issue to correct the ownership structure of this. (Community pool gov module gets direct control of all accounts it controls, e.g. via smart accounts like logic)

It feels like it is a safe assumption that all gov account actions must be dictated by governance. Therefore, the governance module has potential ownership of whatever it interacts with. 

IMO the abstraction should instead be at an even higher level, where this whitelist of gov module should just be expected on every action

File: x/concentrated-liquidity/position_test.go (hunk 2 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/position_test.go:2319]: You are right, thanks :)

**Comment by czarcas7ic** [comment]
Added issue https://github.com/osmosis-labs/osmosis/issues/7778, will likely merge soon

**Pull request merged**